Error on stub trait with type members

Given this trait definition:

```scala
object Network:
  // ...
  trait Effect:
    type Address[_ <: Peer]
    type NetworkError <: Throwable
    type Id

    def getId[P <: Peer](address: Address[P]): Id
    def reachablePeersOf[P <: Peer](peerRepr: PeerRepr): Set[Address[P]]
    // ...
```

Trying to stub that trait as follows

```scala
class PlacedValueTest extends AnyFlatSpecLike, Matchers, Stubs, BeforeAndAfter {
  type IntNetwork = Network.Effect { type Id = Int; type NetworkError = Throwable; type Address[P <: Peer] = String }
  private val netEffect = stub[IntNetwork]
                          ^^^^^^^^^^^^^^^^ java.lang.AssertionError: expected a term symbol, but received val <none>        
  given net: Locicope[IntNetwork](netEffect)

  before:
    resetStubs()
}
```

Fails with:

```
Exception occurred while executing macro expansion.
java.lang.AssertionError: expected a term symbol, but received val <none>
	at org.scalamock.stubs.internal.StubMaker.$anonfun$6(StubMaker.scala:167)
	at scala.collection.immutable.List.flatMap(List.scala:294)
	at org.scalamock.stubs.internal.StubMaker.newInstance(StubMaker.scala:169)
	at org.scalamock.stubs.StubsMacro$package$.stubMacro(StubsMacro.scala:116)
	at org.scalamock.stubs.StubsMacro$package$.inline$stubMacro(StubsMacro.scala:112)
```

--- a/core/shared/src/test/scala-3/com/paulbutcher/test/Scala3Spec.scala
+++ b/core/shared/src/test/scala-3/com/paulbutcher/test/Scala3Spec.scala
@@ -4,8 +4,24 @@ import org.scalamock.scalatest.MockFactory
 import org.scalatest.funspec.AnyFunSpec
 import org.scalatest.matchers.should.Matchers
 
-class   Scala3Spec extends AnyFunSpec with MockFactory with Matchers {
+class Scala3Spec extends AnyFunSpec with MockFactory with Matchers {
 
+  it("mock refinement types") {
+    """
+      |trait Effect:
+      |  type Address[_ <: String]
+      |  type NetworkError <: Throwable
+      |  type Id
+      |
+      |  def getId[P <: String](address: Address[P]): Id
+      |  def reachablePeersOf[P <: String](peerRepr: String): Set[Address[P]]
+      |
+      |  type IntNetwork = Effect {type Id = Int; type NetworkError = Throwable; type Address[P <: String] = String}
+      |
+      |  val netEffect = stub[IntNetwork]
+      |""".stripMargin should compile
+
+  }
 
   it("mock traits with parameters") {
     trait Test(val a: Int) {

